### ISSUE TITLE
Payment Method

### ISSUE THREAD

**Issue by FredericLatour** [paypal/paypal-js#428]
Hi,
I have been trying to find this information for 4 hours without any success.
How can anyone know the payment method that was used once the payment is captured.
The orderData information that is being returned by the capture API doesn't contain any information regarding if the payment was made with a Paypal Account or a Credit Card. I'm really not sure why this information is not provided directly in the response. However, there must be a way . Right ?
Thanks in advance.

**Comment by devchristina** [comment]
Asking the right folks, hope to have an answer for you.

**Comment by FredericLatour** [comment]
@devchristina Any update ?

**Comment by FredericLatour** [comment]
@devchristina Isn't there any person in Paypal who can provide this information?
It's now m 2 weeks !!
There should be a way to know the type of payment used by the customer (Paypal, Credit card, ...).

**Comment by gregjopa** [comment]
Hi @FredericLatour, can you see if `paymentSource` works for you? This value should tell you which button was clicked to start the payment flow: https://github.com/paypal/paypal-js/blob/1acba3f869ed14cd867d3ea33e6657df3d2b820f/types/components/buttons.d.ts#L15

Here's an example:
```js
window.paypal.Buttons({
  createOrder(data, actions) {
    console.log({
      paymentSource: data.paymentSource
    });
  }
})
```

**Comment by FredericLatour** [comment]
Hi @gregjopa 
Thanks for the feedback. 
That's better than nothing but does it mean that PayPal does not keep any information regarding the payment method with a transaction? If for some reason we missed this data ( the paymentSource field on createOrder function) , isn't there any way to retrieve this information for a specific transaction?
That sounds surprising!

**Comment by wsbrunson** [comment]
@FredericLatour you can use the Orders API to get details about the order:
https://developer.paypal.com/docs/api/orders/v2/#orders_get

It should return back the payment source used
```
{
  ...,
  payment_source: {
    [paymentSource]: { ... } // paymentSource will be 'paypal', 'venmo', etc
  }
}
```

**Comment by FredericLatour** [comment]
@wsbrunson 
The problem is there is no distinction between a payment using Paypal Account and a Credit Card.
Retrieving the order after the initial capture doesn't provide more information that what is in the initial capture data.

When I check a transaction through your web interface, I can make the difference between a "Direct Card Payment"
![image](https://github.com/paypal/paypal-js/assets/1407847/8be656bc-a1ef-482a-86b2-81d0c695c2cb)

As opposed to a "Checkout" (Paypal account).
![image](https://github.com/paypal/paypal-js/assets/1407847/ad911988-08b4-486f-8a9b-2046ae6306d1)

How can I discriminate both payment types with certainty (we keep orders in our own database and we obviously want this information). 
There are a couple of fields that seem somewhat different though I'm not sure this is extremely reliable. Paypal should be able to clarify this without ambiguity. 
For instance, when paying with credit card, `Paypal` payment source will be empty as opposed to when paying with a paypal account.
```
payment_source: {
    Paypal: { } 
  }
```
Can we rely on this difference to discriminate between a **Direct Card Payment** and a payment using **Paypal Account**?

Thanks in advance

**Comment by wsbrunson** [comment]
@FredericLatour have you tried to find details about the transaction using our Transaction Search API? https://developer.paypal.com/docs/api/transaction-search/v1/#search_get

**Comment by FredericLatour** [comment]
@wsbrunson Yes, I'm using the [Advanced Checkout approach](https://developer.paypal.com/docs/checkout/advanced/) which corresponds to the unbranded card solution. It seems to be a solution that PayPal is pushing right now.

What specific details about the transaction should we use? Which exact field can we depend on? Wouldn't it be more straightforward if PayPal informed us about the data they use to indicate a 'direct card payment'? This would save us from having to explore the entire API and guess which data might serve this purpose ? :)

**Comment by wsbrunson** [comment]
If you are using the unbranded card feature and want to differentiate between that and when a buyer used PayPal's Checkout, then this message should apply to you:

> @FredericLatour you can use the Orders API to get details about the order: https://developer.paypal.com/docs/api/orders/v2/#orders_get
> 
> It should return back the payment source used
> 
> ```
> {
>   ...,
>   payment_source: {
>     [paymentSource]: { ... } // paymentSource will be 'paypal', 'venmo', etc
>   }
> }
> ```

There should be a single key in the `payment_source` object that will be `paypal` if the transaction was completed with PayPal Checkout and `card` if it was the unbranded.

**Comment by FredericLatour** [comment]
@wsbrunson This is not the case, at least not in the sandbox environment. As I mentioned in a previous post, the 'paymentSource' is 'PayPal' in both scenarios. The only difference is that PayPal account details are present when a payment is made using the PayPal Checkout button (as shown in the sample response on the PayPal developer website), whereas an empty object ({}) is present when the payment is made with a card

PD: the term "unbranded card feature" seems somewhat ambiguous to me.  I'm using the [Advanced Checkout approach](https://developer.paypal.com/docs/checkout/advanced/) which, I suppose, corresponds to the unbranded card solution you are referring to. However, you may want to check in order to ensure this is the case.

**Comment by FredericLatour** [comment]
@wsbrunson Any update on this ?
I just tested an "IDEAL" payment and can confirm that I get "ideal" as the payment source but for direct card payment, this is not the case.

**Comment by wsbrunson** [comment]
what does your order payload look like for paypal vs direct card payment?

**Comment by FredericLatour** [comment]
@wsbrunson I have downloaded and run the 
* [paypaldev/PayPal-React-FullStack-Advanced-Checkout-Sample: Fullstack application with advanced payment processing using the React PayPal SDK.](https://github.com/paypaldev/PayPal-React-FullStack-Advanced-Checkout-Sample)

I connected my sandbox and it looks like I'm getting the proper card information as the "payment source". That's weird.
I will double check and compare both source code in order to see if I can find anything.
Any idea/clue that could facilitate my investigation are welcome.

**Comment by FredericLatour** [comment]
@wsbrunson I was able to pinpoint the origin of the problem. My payload (I understood in the end what you meant by that) was like the following :

```javascript
const payload = {
    intent: 'CAPTURE',
    purchase_units: [
        {
            reference_id: product,
            amount: {
                currency_code: currency,
                value: price.toString(),
            },
            description: product,
        },
    ],
     payment_source: {
         paypal: {
             experience_context: {
                 shipping_preference: 'NO_SHIPPING',
             }
         }
     }
}
```
By removing the `payment_source` property, I am able to get the proper payment type ("card").
The weird thing is that it would not prevent the transaction to properly happen.
Is there a logical explanation why ?

**Comment by wsbrunson** [comment]
> The weird thing is that it would not prevent the transaction to properly happen.
> Is there a logical explanation regarding what's going on here ?

I only work on the SDK so I can't say for certain but my hunch is that

1. When you create the order, you set the payment source as `paypal` and pass that order id to our unbranded card feature
2. When the buyer clicks the submit button with their card information, the unbranded card feature does what we call `confirmPaymentSource` where it patches the current order id with the card details and any billing address information. At that point our backend systems must know to process the transaction as unbranded/Direct Card Payment
3. When retrieving the order, it still shows as `paypal` for the payment source since that's how the order was created.

**Comment by FredericLatour** [comment]
@wsbrunson 

Can you confirm that when using Standard Integration (see picture below), in that case there is no distinction between paying using a CC or a PayPal account (the `payment_source` is paypal in both cases):
![image](https://github.com/paypal/paypal-js/assets/1407847/eb7d1531-798c-446e-8641-3799a5907982)

This is also reflected by the payment type of the resulting transaction :
![image](https://github.com/paypal/paypal-js/assets/1407847/f35a10d4-e399-4c2c-bac7-df2969d535f6)

Granting that my observations are correct, how could I possibly determine if the payment was made using a CC or a PayPal Account ?
